src/vector.rs

```diff
@@ -285,10 +286,20 @@ macro_rules! impl_vector {
     }
 }
 
+macro_rules! impl_vector_scalar_ops {
+    ($($S:ident)*) => ($(
+        impl_scalar_ops!(Vector2<$S> { x, y });
+        impl_scalar_ops!(Vector3<$S> { x, y, z });
+        impl_scalar_ops!(Vector4<$S> { x, y, z, w });
+    )*)
+}
```

Comment:
I would prefer this to be moved into the `impl_vector!` macro, to be consistent with the other codegen. So you would have:
```rust
impl_scalar_ops!($VectorN<usize> { $($field),+ });
impl_scalar_ops!($VectorN<u8> { $($field),+ });
impl_scalar_ops!($VectorN<u16> { $($field),+ });
impl_scalar_ops!($VectorN<u32> { $($field),+ });
impl_scalar_ops!($VectorN<u64> { $($field),+ });
impl_scalar_ops!($VectorN<isize> { $($field),+ });
impl_scalar_ops!($VectorN<i8> { $($field),+ });
impl_scalar_ops!($VectorN<i16> { $($field),+ });
impl_scalar_ops!($VectorN<i32> { $($field),+ });
impl_scalar_ops!($VectorN<i64> { $($field),+ });
impl_scalar_ops!($VectorN<f32> { $($field),+ });
impl_scalar_ops!($VectorN<f64> { $($field),+ });
```
I don't mind the duplication of code there.